include/ada/url_aggregator-inl.h

```diff
@@ -429,8 +428,15 @@ inline void url_aggregator::update_base_password(const std::string_view input) {
 
   add_authority_slashes_if_needed();
 
+  // TODO: Optimization opportunity. Merge the following removal functions.
   if (input.empty()) {
     clear_base_password();
+
+    // Remove username too, if it is empty.
+    if (!has_non_empty_username()) {
+      update_base_username("");
```

Comment:
Why don't we have `clear_base_username()`?

In PR https://github.com/ada-url/ada/pull/308, I am proposing we add ` clear_base_hash()`.

Reply:
I left it as a todo, because this is a really edge case. I prefer coming back to this after fixing the missing edge cases from WPT.